blog/2024-03-28-hybrid-search-typescript/index.mdx

```diff
@@ -0,0 +1,305 @@
+
+A code snippet of Hybrid Search in Javascript will look something like this. 
+
+```typescript
+const myCollection = client.collections.get('JeopardyQuestion');
+
+const result = await myCollection.query.hybrid('food',{
+ limit: 2,
+ returnProperties: ['question', 'answer'],
+})
+
+console.log(JSON.stringify(result.objects, null, 2));
+```
+
+
+## Optimizing your Hybrid Search
+
+With Weaviates Hybrid Search, we have two ways to adjust the Hybrid Search algorithm. We can adjust the weights by tweaking our search alpha or changing our ranking algorithm.
+
+### Changing alpha
+
+Think of alpha as a measure of how much your hybrid search should lean towards either Vector Search or keyword search where an alpha of 1 is a pure vector search, and an alpha of 0 is a pure keyword search. It is 0.75 by default. 
+
+
+
+### Changing ranking algorithm 
+
+Here is where it gets fun! Weaviate supports two fusion algorithms as of Weaviate v1.20 i.e. rankedFusion and relativeScoreFusion. 
+
+In a nutshell, rankedFusion (the original Hybrid search algorithm in Weaviate) assigns each object a score based on its position in the search results (in descending order), with the total score derived from the sum of these rank-based scores across both vector and keyword searches.
+
+![Ranked Fusion Algorithm](./img/reciprocal-rank.png)
+
+
+Relative Score Fusion on the other hand normalizes the output of the vector and keyword searches so that the highest value becomes 1 and the lowest 0 and everything in between is scaled to match. The total score is then the sum of the normalized individual search scores. Letting us maintain the relevance given by the original keyword/vector search algorithms, while still being able to compare the results and fuse them. As relativeScoreFusion is the newer algorithm, it would most likely be the better choice for most use cases.
+
+![Reciprocal Score Fusion Algorithm](./img/relative-score.png)
+
+
+Dirk and Joon-Pil on our team wrote an [in-depth blog post](https://weaviate.io/blog/hybrid-search-fusion-algorithms) on Hybrid Search in Weaviate. I would recommend looking at it for an in-depth look. 
+
+
+
+## When to use Hybrid Search
+
+So with all that you know about Hybrid Search, when should you use it? 
+
+The way we see it, if you have a model fine-tuned for your data, pure vector search should be enough to give you accurate search results. If you’re mostly matching patterns that rely on keyword matching, keyword search should be good enough. 
+
+If your use cases don't fall into any of the two scenarios mentioned, you might benefit from implementing hybrid search. It isn’t a silver bullet as you would have to optimize it for your data with the customization options we have available. It’s worth noting that because you’re running two searches in the background and computing your results from them, there is a performance cost associated with each search for high-throughput applications.
```

Comment:
> It isn’t a silver bullet as you would have to optimize it for your data with the customization options we have available.

Not sure what you mean by this.

Reply:
I mean to say it's implementation takes some a customization to suit your use case